group-actors/src/main/java/org/sunbird/actors/ReadGroupActor.java

```diff
@@ -35,15 +39,36 @@ public void onReceive(Request request) throws Throwable {
    *
    * @param actorMessage
    */
-  private void readGroup(Request actorMessage) throws BaseException {
+  private void readGroup(Request actorMessage) throws Exception {
     logger.info("ReadGroup method call");
     String groupId = (String) actorMessage.getRequest().get(JsonKey.GROUP_ID);
     List<String> requestFields = (List<String>) actorMessage.getRequest().get(JsonKey.FIELDS);
-
+    GroupResponse groupResponse;
+    String groupInfo = cacheUtil.getCache(groupId);
+    if (StringUtils.isNotEmpty(groupInfo)) {
+      groupResponse = JsonUtils.deserialize(groupInfo, GroupResponse.class);
```

Comment:
I see deserialization here and then in 70 to convert to a Map. Can we do one thing only - at least for Redis cache reads?

Reply:
Yes, readGroup method returns group object, then I am converting into a string and saving into cache. But to put in the  response, it should be a MAp<String , Object> I cannot add groupObject so I am converting into Map at 70